### ISSUE TITLE
Shared OAuth refresh token not accepted

### ISSUE THREAD

**Issue by dkocher** [iterate-ch/cyberduck#15284]
From #15230. OAuth tokens saved in the credentials manager shared between _Cyberduck_ and _Mountain Duck_ seem to cause issues for OneDrive although using the same registered OAuth Client ID for _Cyberduck_ and _Mountain Duck_.

```
AADSTS70000: The provided value for the input parameter 'refresh_token' or 'assertion' is not valid. The token was issued for a different client id
```

**Comment by AliveDevil** [comment]
> using the same registered OAuth Client ID for Cyberduck and Mountain Duck
This is false. OAuth Client Ids differ for
- OneDrive
- Google Drive
- Google Cloud Storage
- Dropbox.

**Comment by dkocher** [comment]
Related issue in #15309.

**Comment by dkocher** [comment]
Will be implemented as part of #15114.
